### ISSUE TITLE
Latest update broke @with_kw inside a module

### ISSUE THREAD

**Issue by cstjean** [mauro3/Parameters.jl#5]
This fails:
```julia
module PTest

using Parameters: @with_kw

@with_kw type Foo
    x
end

end
```

```
> import PTest
LoadError: LoadError: UndefVarError: Parameters not defined
while loading PTest.jl, in expression starting on line 17
while loading In[4], in expression starting on line 1
```

It works fine interactively though. It's just if I put it inside a module that I get the error.

I've bisected it:
```
acc7fe642c728a02a027d3c85352575c2e40f2ae is the first bad commit
commit acc7fe642c728a02a027d3c85352575c2e40f2ae
Author: Mauro Werder <mauro_lc@runbox.com>
Date:   Mon Nov 23 22:08:26 2015 +0100

    Fixed #4

:040000 040000 f4a06ecc93b8bf62bc3df7bce3a21820e42d1ce4 af7b2cbdabedc65252a7e8f8bc76a52657108535 M	src
:040000 040000 e440f2915ff003c8cf0a8492ef3e573007a5578f 51a7f27b743e5a4229c6b9beb8b67addde044ee3 M	test
```
It's probably [this line](https://github.com/mauro3/Parameters.jl/commit/acc7fe642c728a02a027d3c85352575c2e40f2ae#diff-ec1155cc07b8ae5fc6787c9cb23a7986R298) that causes the issue.

**Comment by mauro3** [comment]
Thanks for reporting this.  Looks quite odd to me.  It works like so:
```julia
module PTest
using Parameters
@with_kw type Foo
    x
end
end
```
or
```julia
module pa
import Parameters
Parameters.@with_kw type Foo
    x
end
end
```
I suspect this is a Julia bug (always blame the others first :-)

**Comment by mauro3** [comment]
It wasn't Julia's fault after all... Using `using Parameters: @with_kw` will only bind `Parameters` itself into the `Main` module and not into the module inside which that statement was made.  However, I relied on `Parameters` being bound in the module scope, thus it failed. (I think I may have used this wrong assumption in other macros I programmed, good to know).

**Comment by cstjean** [comment]
Interesting behaviour. Thank you for the quick response!

**Comment by mauro3** [comment]
Keep them coming... Feature requests/general comments also!

**Comment by mauro3** [comment]
There were more instances of this bug in the code, for example:
```julia
module A
       using Parameters: @pack
       immutable AA
           a
       end
       aa = AA(1)
       a = 1
       @pack aa: a
end
```
I think they are all fixed in commit 639202c, so if they bite again you need to checkout master.
